tensorflow/compiler/xla/service/cpu/onednn_matmul.cc

```diff
@@ -0,0 +1,88 @@
+/* Copyright 2023 The TensorFlow Authors. All Rights Reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+==============================================================================*/
+#if defined(INTEL_MKL) && defined(ENABLE_ONEDNN_V3)
+
+#include "tensorflow/compiler/xla/service/cpu/onednn_matmul.h"
+
+#include <algorithm>
+#include <cmath>
+#include <initializer_list>
+#include <vector>
+
+#include "absl/base/dynamic_annotations.h"
+#include "dnnl.hpp"
+#include "tensorflow/compiler/xla/executable_run_options.h"
+#include "tensorflow/compiler/xla/service/cpu/backend_config.pb.h"
+#include "tensorflow/compiler/xla/service/cpu/onednn_memory_util.h"
+#include "tensorflow/compiler/xla/service/cpu/runtime_lightweight_check.h"
+#include "tensorflow/tsl/util/onednn_threadpool.h"
+#include "third_party/eigen3/unsupported/Eigen/CXX11/Tensor"
+
+namespace xla {
+namespace cpu {
+
+using namespace dnnl;
+
+ABSL_ATTRIBUTE_NO_SANITIZE_MEMORY void __xla_cpu_runtime_OneDnnMatMul(
+    const void* run_options_ptr, void* lhs, void* rhs, void* result,
+    void* config) {
+  const xla::ExecutableRunOptions* run_options =
+      static_cast<const xla::ExecutableRunOptions*>(run_options_ptr);
+  XLA_LIGHTWEIGHT_CHECK(run_options != nullptr);
+  XLA_LIGHTWEIGHT_CHECK(run_options->intra_op_thread_pool() != nullptr);
+  // TODO(inte-tf): Update the namespace scope of threadpool once the
+  // threadpool interface wrapper is moved as tsl::OneDnnThreadPool.
```

Comment:
This comment can be removed. The threadpool interface is already `tsl::OneDnnThreadPool`.